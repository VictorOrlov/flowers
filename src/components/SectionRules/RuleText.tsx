import * as React from "react";

const RuleText: React.FC = () => (
  <>
    <div className="col-md-10 rules">
      <h2 className="section_title">ДАВАЙТЕ УКРАСИМ КАЗАНЬ ВМЕСТЕ!</h2>
      <p className="section_text">
        Преследуя важную задачу вовлечения населения в процесс благоустройства и
        озеленения нашего города, мы приглашаем Вас принять участие в конкурсе
        «Цветущая Казань» и вместе с нами преобразить и украсить нашу столицу.
      </p>
      <p className="section_text">
        Вам предоставляется возможность своими руками, проявив фантазию и
        воображение, создать красивую фасадную композицию, будь то клумба,
        палисадник или цветочное оформление территории. Также Вы можете
        применить свое фото-мастерство и оригинальность и участвовать в данном
        конкурсе в категории «Лучший цветочный кадр».
      </p>
    </div>
    <div className="col-md-10 rules">
      <h2 className="section_title">ПРАВИЛА КОНКУРСА</h2>
      <p className="section_text">
        Из перечня всех предлагаемых номинаций, находящихся на нашем сайте, Вы
        сможете выбрать любую, которая Вам по душе и непосредственно подходящую
        для Ваших условий, места проживания и территориального расположения.
        Далее Вам будет необходимо выполнить соответствующие действия для
        создания композиции или цветочного оформления, сделать несколько
        фотографий своего готового объекта и прикрепить к{" "}
        <a href="/#section-send-request">заявке участника</a>, которую также
        можно найти на сайте нашего конкурса. После одобрения конкурсной
        комиссией заявки, за неё можно будет проголосовать, но вносить в неё
        изменения будет уже невозможно.
      </p>
      <p className="section_text">
        Оценка участников будет осуществляется посредством народного
        голосования, главным жюри которого станут жители и гости города Казань.
        Вы сможете отслеживать количество набранных Вами голосов и всегда быть в
        курсе происходящего.
      </p>
      <p className="section_text">
        Стоит отметить, что участниками Конкурса, помимо жителей Казани, могут
        стать разного рода предприятия, организации, а также учреждения
        различных организационно-правовых форм. Словом – все те, кто хочет
        внести свой бесценный вклад в улучшение внешнего облика нашего города.
      </p>
      <p className="section_text">
        Подводить итоги голосования будет городская конкурсная комиссия, которая
        и определит впоследствии победителей конкурса. Авторы лучших работ
        станут обладателями ценных призов и подарков. Желаем успехов!
      </p>
    </div>
  </>
);

export default RuleText;
