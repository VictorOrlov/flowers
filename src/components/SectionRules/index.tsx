import * as React from "react";
import { Competition } from "../../graphql";
import RulesText from "./RuleText";
// import shedule from "../../images/shedule_svg.svg";

interface IProps {
  competitions: Array<Competition>;
}

const SectionRules: React.FunctionComponent<IProps> = props => {
  const renderCompetitions = props.competitions.map(competition => (
    <div className="col-md-6 rules_nominations_wrapper" key={competition.id}>
      <span className="nominations_list">{competition.name}</span>
    </div>
  ));
  return (
    <section>
      <RulesText />
      <div className="row rules_shedule">
        <div className="col-md-10 shedule_svg">
          <img
            src="../../images/shedule_svg.svg"
            alt="shedule"
            className="img-fluid"
          />
        </div>
      </div>
      <div className="row rules_nominations">
        <div className="col-md-10 nominations">
          <h3 className="nomination_subtitle">
            Конкурс проводится по следующим номинациям:
          </h3>
          <div className="row">{renderCompetitions}</div>
        </div>
        <hr />
      </div>
      <div className="row rules_promo">
        <div className="col-md-6 promo">
          <h3>
            Оценка участников Конкурса по соответствующим номинациям
            осуществляется народным голосованием среди жителей и гостей города
            Казань
          </h3>
        </div>
      </div>
    </section>
  );
};

export default SectionRules;
