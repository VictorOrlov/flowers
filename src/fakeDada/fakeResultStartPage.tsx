export const fakeResultStartPage = {
  data: {
    startPage: {
      pagesCount: 22,
      requests: [
        {
          address: "Все Унистроится!",
          comment: null,
          competition: {
            id: 8,
            kind: "close",
            name: "Цветочный фестиваль"
          },
          description: null,
          id: 399,
          participant: {
            id: 360,
            kind: "organization",
            name: "Девелоперская компания Унистрой",
            user: null
          },
          photos: [
            {
              id: 2516
            },
            {
              id: 2517
            }
          ],
          state: "approved",
          voicesCount: 20,
          createdAt: "2018-07-04 09:24:34 UTC",
          year: 2019
        },
        {
          address: "Г.Казань, ул.Шамиля Усманова, 37 а",
          comment: null,
          competition: {
            id: 7,
            kind: "open",
            name:
              "Лучшее оформление территории, прилегающей к образовательному учреждению"
          },
          description: null,
          id: 369,
          participant: {
            id: 338,
            kind: "person",
            name: null,
            user: {
              id: 677,
              email: "ds222.kzn@tatar.ru",
              name: "Светлана",
              phone: "+79377791511",
              remainVoices: null,
              roles: ["member"],
              surname: "Гафиятуллина"
            }
          },
          photos: [
            {
              id: 2258
            },
            {
              id: 2259
            },
            {
              id: 2260
            },
            {
              id: 2261
            },
            {
              id: 2262
            },
            {
              id: 2263
            }
          ],
          state: "approved",
          voicesCount: 17,
          createdAt: "2018-06-26 08:55:03 UTC",
          year: 2019
        },
        {
          address: "г Казань, пр-кт Победы, д 180А",
          comment: null,
          competition: {
            id: 10,
            kind: "open",
            name: "Лучший цветочный кадр в городской среде"
          },
          description: null,
          id: 534,
          participant: {
            id: 472,
            kind: "person",
            name: null,
            user: {
              id: 60458,
              email: "ramis1707@mail.ru",
              name: "Эльмира",
              phone: "+79196233739",
              remainVoices: null,
              roles: [],
              surname: "Абдрахманова"
            }
          },
          photos: [
            {
              id: 3288
            },
            {
              id: 3297
            }
          ],
          state: "approved",
          voicesCount: 17,
          createdAt: "2018-08-03 13:15:30 UTC",
          year: 2019
        }
      ],
      winnersPreviousYear: []
    }
  }
};
