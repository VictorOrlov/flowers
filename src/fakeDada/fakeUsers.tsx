export const UserApproved = {
  email: "lol.tebeb@n.ru",
  name: "Одобрен",
  surname: "Зубенко",
  id: 258,
  participant: {
    name: 'ООО УК "ЖКХ ДЕРБЫШКИ"',
    request: {
      address: "Батурина 18п",
      competition: {
        name: "Лучшие цветы на районе"
      },
      participant: {
        name: 'ООО УК "ЖКХ ДЕРБЫШКИ"'
      },
      description: "Почему белые розы похожи на пелемени?",
      photos: [
        {
          id: 1,
          address:
            "https://i.pinimg.com/originals/e7/a9/a1/e7a9a19464b5768887c4dbb9b0b09119.gif"
        },
        {
          id: 2,
          address:
            "http://cloudmind.info/wp-content/uploads/2014/12/blue-water-lily-flower-flowers-hd-wallpaper-beautiful-gallery-hd1.jpg"
        },
        {
          id: 3,
          address:
            "https://www.madeira-web.com/images/latest-news/festaflor19-a.jpg"
        },
        {
          id: 4,
          address:
            "http://cloudmind.info/wp-content/uploads/2014/12/blue-water-lily-flower-flowers-hd-wallpaper-beautiful-gallery-hd1.jpg"
        }
      ],
      state: "approved"
    }
  },
  phone: "8-987-2222-33-44",
  remainVoices: 7,
  voices: [117, 54, 200, 693],
  roles: ["member"]
};

export const UserUpdated = {
  email: "lol.tebeb@n.ru",
  name: "Обновил",
  surname: "Зубенко",
  id: 258,
  participant: {
    name: 'ООО УК "ЖКХ ДЕРБЫШКИ"',
    request: {
      address: "Батурина 18п",
      competition: {
        name: "Лучшие цветы на районе"
      },
      participant: {
        name: 'ООО УК "ЖКХ ДЕРБЫШКИ"'
      },
      description: "Почему белые розы похожи на пелемени?",
      photos: [
        {
          id: 1,
          address:
            "https://i.pinimg.com/originals/e7/a9/a1/e7a9a19464b5768887c4dbb9b0b09119.gif"
        },
        {
          id: 2,
          address:
            "http://cloudmind.info/wp-content/uploads/2014/12/blue-water-lily-flower-flowers-hd-wallpaper-beautiful-gallery-hd1.jpg"
        },
        {
          id: 3,
          address:
            "https://www.madeira-web.com/images/latest-news/festaflor19-a.jpg"
        },
        {
          id: 4,
          address:
            "http://cloudmind.info/wp-content/uploads/2014/12/blue-water-lily-flower-flowers-hd-wallpaper-beautiful-gallery-hd1.jpg"
        }
      ],
      state: "updated"
    }
  },
  phone: "8-987-2222-33-44",
  remainVoices: 7,
  voices: [117, 54, 200, 693],
  roles: ["member"]
};

export const UserDeclined = {
  email: "lol.tebeb@n.ru",
  name: "Отклонён",
  surname: "Зубенко",
  id: 258,
  participant: {
    name: 'ООО УК "ЖКХ ДЕРБЫШКИ"',
    request: {
      address: "Батурина 18п",
      competition: {
        name: "Лучшие цветы на районе"
      },
      participant: {
        name: 'ООО УК "ЖКХ ДЕРБЫШКИ"'
      },
      description: "Почему белые розы похожи на пелемени?",
      photos: [
        {
          id: 1,
          address:
            "https://i.pinimg.com/originals/e7/a9/a1/e7a9a19464b5768887c4dbb9b0b09119.gif"
        },
        {
          id: 2,
          address:
            "http://cloudmind.info/wp-content/uploads/2014/12/blue-water-lily-flower-flowers-hd-wallpaper-beautiful-gallery-hd1.jpg"
        },
        {
          id: 3,
          address:
            "https://www.madeira-web.com/images/latest-news/festaflor19-a.jpg"
        },
        {
          id: 4,
          address:
            "http://cloudmind.info/wp-content/uploads/2014/12/blue-water-lily-flower-flowers-hd-wallpaper-beautiful-gallery-hd1.jpg"
        }
      ],
      state: "declined"
    }
  },
  phone: "8-987-2222-33-44",
  remainVoices: 7,
  voices: [117, 54, 200, 693],
  roles: ["member"]
};

export const UserCreated = {
  email: "lol.tebeb@n.ru",
  name: "Создал",
  surname: "Зубенко",
  id: 258,
  participant: {
    name: 'ООО УК "ЖКХ ДЕРБЫШКИ"',
    request: {
      address: "Батурина 18п",
      competition: {
        name: "Лучшие цветы на районе"
      },
      participant: {
        name: 'ООО УК "ЖКХ ДЕРБЫШКИ"'
      },
      description: "Почему белые розы похожи на пелемени?",
      photos: [
        {
          id: 1,
          address:
            "https://i.pinimg.com/originals/e7/a9/a1/e7a9a19464b5768887c4dbb9b0b09119.gif"
        },
        {
          id: 2,
          address:
            "http://cloudmind.info/wp-content/uploads/2014/12/blue-water-lily-flower-flowers-hd-wallpaper-beautiful-gallery-hd1.jpg"
        },
        {
          id: 3,
          address:
            "https://www.madeira-web.com/images/latest-news/festaflor19-a.jpg"
        },
        {
          id: 4,
          address:
            "http://cloudmind.info/wp-content/uploads/2014/12/blue-water-lily-flower-flowers-hd-wallpaper-beautiful-gallery-hd1.jpg"
        }
      ],
      state: "created"
    }
  },
  phone: "8-987-2222-33-44",
  remainVoices: 7,
  voices: [117, 54, 200, 693],
  roles: ["member"]
};

export const UserDeleted = {
  email: "lol.tebeb@n.ru",
  name: "УдалёнНаУчастие",
  surname: "Зубенко",
  id: 258,
  participant: {
    name: 'ООО УК "ЖКХ ДЕРБЫШКИ"',
    request: {
      address: "Батурина 18п",
      competition: {
        name: "Лучшие цветы на районе"
      },
      participant: {
        name: 'ООО УК "ЖКХ ДЕРБЫШКИ"'
      },
      description: "Почему белые розы похожи на пелемени?",
      photos: [
        {
          id: 1,
          address:
            "https://i.pinimg.com/originals/e7/a9/a1/e7a9a19464b5768887c4dbb9b0b09119.gif"
        },
        {
          id: 2,
          address:
            "http://cloudmind.info/wp-content/uploads/2014/12/blue-water-lily-flower-flowers-hd-wallpaper-beautiful-gallery-hd1.jpg"
        },
        {
          id: 3,
          address:
            "https://www.madeira-web.com/images/latest-news/festaflor19-a.jpg"
        },
        {
          id: 4,
          address:
            "http://cloudmind.info/wp-content/uploads/2014/12/blue-water-lily-flower-flowers-hd-wallpaper-beautiful-gallery-hd1.jpg"
        }
      ],
      state: "deleted"
    }
  },
  phone: "8-987-2222-33-44",
  remainVoices: 7,
  voices: [117, 54, 200, 693],
  roles: ["member"]
};

export const UserDeletedForNewRequest = {
  email: "lol.tebeb@n.ru",
  name: "Удалён но в игре",
  surname: "Зубенко",
  id: 258,
  participant: {
    name: 'ООО УК "ЖКХ ДЕРБЫШКИ"',
    request: {
      address: "Батурина 18п",
      competition: {
        name: "Лучшие цветы на районе"
      },
      participant: {
        name: 'ООО УК "ЖКХ ДЕРБЫШКИ"'
      },
      description: "Почему белые розы похожи на пелемени?",
      photos: [
        {
          id: 1,
          address:
            "https://i.pinimg.com/originals/e7/a9/a1/e7a9a19464b5768887c4dbb9b0b09119.gif"
        },
        {
          id: 2,
          address:
            "http://cloudmind.info/wp-content/uploads/2014/12/blue-water-lily-flower-flowers-hd-wallpaper-beautiful-gallery-hd1.jpg"
        },
        {
          id: 3,
          address:
            "https://www.madeira-web.com/images/latest-news/festaflor19-a.jpg"
        },
        {
          id: 4,
          address:
            "http://cloudmind.info/wp-content/uploads/2014/12/blue-water-lily-flower-flowers-hd-wallpaper-beautiful-gallery-hd1.jpg"
        }
      ],
      state: "deleted_for_new_request"
    }
  },
  phone: "8-987-2222-33-44",
  remainVoices: 7,
  voices: [117, 54, 200, 693],
  roles: ["member"]
};

export const UserNoParticipation = {
  email: "lol.tebeb@n.ru",
  name: "Не учавствую",
  surname: "Пфф",
  id: 258,
  participant: null,
  phone: "8-987-2222-33-44",
  remainVoices: 4,
  voices: [117, 54, 200, 693],
  roles: ["member"]
};

export const UserNotAuth = null;

export const UserRemainVoicesEnded = {
  email: "lol.tebeb@n.ru",
  name: "Нет голосов",
  surname: "КакТак",
  id: 258,
  participant: {
    name: 'ООО Нет Голосов"',
    request: {
      address: "Транжира 18п",
      competition: {
        name: "Лучшие цветы на районе"
      },
      participant: {
        name: "ООО Нет Голосов"
      },
      description: "Почему белые розы похожи на пелемени?",
      photos: [
        {
          id: 1,
          address:
            "https://i.pinimg.com/originals/e7/a9/a1/e7a9a19464b5768887c4dbb9b0b09119.gif"
        },
        {
          id: 2,
          address:
            "http://cloudmind.info/wp-content/uploads/2014/12/blue-water-lily-flower-flowers-hd-wallpaper-beautiful-gallery-hd1.jpg"
        },
        {
          id: 3,
          address:
            "https://www.madeira-web.com/images/latest-news/festaflor19-a.jpg"
        },
        {
          id: 4,
          address:
            "http://cloudmind.info/wp-content/uploads/2014/12/blue-water-lily-flower-flowers-hd-wallpaper-beautiful-gallery-hd1.jpg"
        }
      ],
      state: "deleted_for_new_request"
    }
  },
  phone: "8-987-2222-33-44",
  remainVoices: 0,
  voices: [117, 54, 200, 693],
  roles: ["member"]
};

export const UserVotedThisRequest = {
  email: "lol.tebeb@n.ru",
  name: "Нет голосов",
  surname: "КакТак",
  id: 258,
  participant: {
    name: 'ООО Нет Голосов"',
    request: {
      address: "Транжира 18п",
      competition: {
        name: "Лучшие цветы на районе"
      },
      participant: {
        name: "ООО Нет Голосов"
      },
      description: "Почему белые розы похожи на пелемени?",
      photos: [
        {
          id: 1,
          address:
            "https://i.pinimg.com/originals/e7/a9/a1/e7a9a19464b5768887c4dbb9b0b09119.gif"
        },
        {
          id: 2,
          address:
            "http://cloudmind.info/wp-content/uploads/2014/12/blue-water-lily-flower-flowers-hd-wallpaper-beautiful-gallery-hd1.jpg"
        },
        {
          id: 3,
          address:
            "https://www.madeira-web.com/images/latest-news/festaflor19-a.jpg"
        },
        {
          id: 4,
          address:
            "http://cloudmind.info/wp-content/uploads/2014/12/blue-water-lily-flower-flowers-hd-wallpaper-beautiful-gallery-hd1.jpg"
        }
      ],
      state: "deleted_for_new_request"
    }
  },
  phone: "8-987-2222-33-44",
  remainVoices: 4,
  voices: [
    390,
    609,
    559,
    599,
    627,
    593,
    536,
    611,
    544,
    415,
    497,
    408,
    551,
    554,
    463,
    633,
    598,
    117,
    54,
    200,
    693,
    629,
    653,
    545,
    665,
    415,
    688,
    470,
    520,
    433,
    542,
    277,
    278,
    357
  ],
  roles: ["member"]
};

//////////////Types//////////////Types//////////////Types//////////////Types//////////////Types//////////////
type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
};

export type FakeUserType = {
  email: string;
  name: string;
  surname: string;
  id: number;
  participant?: FakeParticipant;
  phone: string;
  remainVoices: number;
  voices: Array<Scalars["Int"]>;
  roles: Array<Scalars["String"]>;
};

type FakeParticipant = {
  name: Scalars["String"];
  request?: FakeRequest;
};

export type FakeRequest = {
  address: Scalars["String"];
  comment?: Scalars["String"];
  competition: Competition;
  description?: Scalars["String"];
  id?: Scalars["Int"];
  participant: FakeParticipant;
  photos: Array<Photo>;
  state: Scalars["String"];
  position?: Array<Scalars["Int"]>;
};

type Competition = {
  name: Scalars["String"];
};

type Photo = {
  id: Scalars["Int"];
  address: Scalars["String"];
};
