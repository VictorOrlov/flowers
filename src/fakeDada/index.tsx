export const fakeCompititions = [
  {
    id: 3,
    name: "Лучшее оформление фасада",
    kind: "open"
  },
  {
    id: 4,
    name: "Лучшее оформление дворовой территории",
    kind: "open"
  },
  {
    id: 5,
    name:
      "Лучшее оформление прилегающей территории для организаций и предприятий",
    kind: "open"
  },
  {
    id: 6,
    name: "Самая оригинальная цветочная композиция",
    kind: "open"
  },
  {
    id: 7,
    name:
      "Лучшее оформление территории, прилегающей к образовательному учреждению",
    kind: "open"
  },
  {
    id: 10,
    name: "Лучший цветочный кадр в городской среде",
    kind: "open"
  }
];

export const fakeWinnersPreviousYear = [
  {
    competition: {
      name: "Лучшие цветы на районе"
    },
    id: 1,
    place: 1,
    request: {
      address: "Гагарина",
      voicesCount: 111,
      photos: [
        {
          id: 76,
          url:
            "https://travel.home.sndimg.com/content/dam/images/travel/fullset/2012/04/03/51/flower-festivals_ss_001.rend.hgtvcom.966.725.suffix/1491582683953.jpeg"
        },
        {
          id: 77,
          url:
            "https://images.rove.me/w_1920,q_85/ysku69nzck3nqudnfspp/madeira-madeira-flower-festival.jpg"
        },
        {
          id: 78,
          url:
            "https://d3h30waly5w5yx.cloudfront.net/images/tour/pictures/taean-tulip-festival.jpg"
        },
        {
          id: 176,
          url:
            "https://www.thaifestivalblogs.com/wp-content/uploads/2018/12/flowerfest-1.jpg"
        },
        {
          id: 177,
          url:
            "https://www.saga.co.uk/contentlibrary/saga/publishing/verticals/travel/world-travel-events/madeira-flower-festival.jpg"
        }
      ]
    }
  },
  {
    competition: {
      name: "Лучшие цветочная композиция"
    },
    id: 2,
    place: 2,
    request: {
      address: "Комарова",
      voicesCount: 98,
      photos: [
        {
          id: 176,
          url:
            "https://www.thaifestivalblogs.com/wp-content/uploads/2018/12/flowerfest-1.jpg"
        },
        {
          id: 177,
          url:
            "https://www.saga.co.uk/contentlibrary/saga/publishing/verticals/travel/world-travel-events/madeira-flower-festival.jpg"
        },
        {
          id: 178,
          url: "https://i.ytimg.com/vi/RsO3slPCMrc/maxresdefault.jpg"
        }
      ]
    }
  },
  {
    competition: {
      name: "Цветочный фестиваль"
    },
    id: 3,
    place: 3,
    request: {
      address: "Короленко",
      voicesCount: 55,
      photos: [
        {
          id: 276,
          url:
            "https://www.madeira-web.com/images/latest-news/festaflor19-a.jpg"
        },
        {
          id: 277,
          url:
            "https://www.salisburycathedral.org.uk/sites/default/files/018%20Magna%20Flora%20Flower%20Festival%20-%20by%20Ash%20Mills.jpg"
        },
        {
          id: 278,
          url:
            "http://indochinatravelblog.com/wp-content/uploads/2015/12/festival_EWUL.jpg"
        }
      ]
    }
  }
];

export const fakeRequestsForMap = [
  {
    id: 543,
    address: "Гагарина 111",
    participant: {
      name: 'ООО Мы рядом с Урицким"'
    },
    position: [55.836458, 49.071035]
  },
  {
    id: 654,
    address: "Восстания, 49",
    participant: {
      name: 'Отдел Полиции"'
    },
    position: [55.832727, 49.076439]
  },
  {
    id: 441,
    address: "Батурина, 18п",
    participant: {
      name: 'ООО УК "ЖКХ ДЕРБЫШКИ"'
    },
    position: [55.798372, 49.109653]
  },
  {
    id: 474,
    address: " Петербургская, 57",
    participant: {
      name: 'ДЦ "Экият"'
    },
    position: [55.7798, 49.138061]
  },
  {
    id: 399,
    address: "Ямашева 56",
    participant: {
      name: "МакДак"
    },
    position: [55.826742, 49.155025]
  },
  {
    id: 598,
    address: "Глазунова 8",
    participant: {
      name: "ООО ПростоДом"
    },
    position: [55.733645, 49.14532]
  },
  {
    id: 586,
    address: "Зеленодольск, парк Авангард",
    participant: {
      name: "Парк Авангард"
    },
    position: [55.848135, 48.507141]
  }
];

///////////////////////////////TYPES//////////////////////////////////

type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
};

type Competition = {
  name: Scalars["String"];
};

type Photo = {
  id: Scalars["Int"];
  address: Scalars["String"];
};

export type FakeRequest = {
  address: Scalars["String"];
  comment?: Scalars["String"];
  competition: Competition;
  description?: Scalars["String"];
  id?: Scalars["Int"];
  participant: FakeParticipant;
  photos: Array<Photo>;
  state: Scalars["String"];
  position?: Array<Scalars["Int"]>;
};

type FakeParticipant = {
  name: Scalars["String"];
  request?: FakeRequest;
};

// "jest": {
//   "roots": [
//     "<rootDir>/src"
//   ],
//   "collectCoverageFrom": [
//     "src/**/*.{js,jsx,ts,tsx}",
//     "!src/**/*.d.ts"
//   ],
//   "setupFiles": [
//     "react-app-polyfill/jsdom"
//   ],
//   "setupFilesAfterEnv": [
//     "<rootDir>/src/setupTests.ts",
//     "./node_modules/jest-enzyme/lib/index.js"
//   ],
//   "testMatch": [
//     "<rootDir>/src/**/__tests__/**/*.{js,jsx,ts,tsx}",
//     "<rootDir>/src/**/*.{spec,test}.{js,jsx,ts,tsx}"
//   ],
//   "preset": "ts-jest",
//   "testEnvironment": "jest-environment-jsdom-fourteen",
//   "transform": {
//     "^.+\\.(js|jsx|ts|tsx)$": "<rootDir>/node_modules/babel-jest",
//     "^.+\\.css$": "<rootDir>/config/jest/cssTransform.js",
//     "^(?!.*\\.(js|jsx|ts|tsx|css|json)$)": "<rootDir>/config/jest/fileTransform.js"
//   },
//   "transformIgnorePatterns": [
//     "[/\\\\]node_modules[/\\\\].+\\.(js|jsx|ts|tsx)$",
//     "^.+\\.module\\.(css|sass|scss)$"
//   ],
//   "modulePaths": [],
//   "moduleNameMapper": {
//     "@/(.*)$": "<rootDir>/src/$1",
//     "^react-native$": "react-native-web",
//     "^.+\\.module\\.(css|sass|scss)$": "identity-obj-proxy"
//   },
//   "moduleFileExtensions": [
//     "web.js",
//     "js",
//     "web.ts",
//     "svg",
//     "ts",
//     "web.tsx",
//     "tsx",
//     "json",
//     "web.jsx",
//     "jsx",
//     "node"
//   ],
//   "watchPlugins": [
//     "jest-watch-typeahead/filename",
//     "jest-watch-typeahead/testname"
//   ],
//   "snapshotSerializers": [
//     "enzyme-to-json/serializer"
//   ],
//   "setupTestFrameworkScriptFile": "<rootDir>/src/setupTests.ts"
// },
