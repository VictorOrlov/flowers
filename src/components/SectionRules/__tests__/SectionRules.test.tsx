import React from 'react';
import { mount } from 'enzyme';
import SectionRules from '..';
import { fakeCompititions } from '../../../fakeDada';

const component = mount(
  <SectionRules competitions={fakeCompititions}/>
)

it ('displaying data prop in SectionRules', ()=>{
  expect(component.find('.row').at(2).childAt(0).text()).toBe(fakeCompititions[0].name);
  expect(component.find('.row').at(2).childAt(5).text()).toBe(fakeCompititions[5].name);
});

it("snapshot SectionRules Component", ()=>{
  expect(component).toMatchSnapshot();
});