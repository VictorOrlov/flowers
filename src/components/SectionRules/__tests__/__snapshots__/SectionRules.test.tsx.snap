// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot SectionRules Component 1`] = `
<SectionRules
  competitions={
    Array [
      Object {
        "id": 3,
        "kind": "open",
        "name": "Лучшее оформление фасада",
      },
      Object {
        "id": 4,
        "kind": "open",
        "name": "Лучшее оформление дворовой территории",
      },
      Object {
        "id": 5,
        "kind": "open",
        "name": "Лучшее оформление прилегающей территории для организаций и предприятий",
      },
      Object {
        "id": 6,
        "kind": "open",
        "name": "Самая оригинальная цветочная композиция",
      },
      Object {
        "id": 7,
        "kind": "open",
        "name": "Лучшее оформление территории, прилегающей к образовательному учреждению",
      },
      Object {
        "id": 10,
        "kind": "open",
        "name": "Лучший цветочный кадр в городской среде",
      },
    ]
  }
>
  <section>
    <RuleText>
      <div
        className="col-md-10 rules"
      >
        <h2
          className="section_title"
        >
          ДАВАЙТЕ УКРАСИМ КАЗАНЬ ВМЕСТЕ!
        </h2>
        <p
          className="section_text"
        >
          Преследуя важную задачу вовлечения населения в процесс благоустройства и озеленения нашего города, мы приглашаем Вас принять участие в конкурсе «Цветущая Казань» и вместе с нами преобразить и украсить нашу столицу.
        </p>
        <p
          className="section_text"
        >
          Вам предоставляется возможность своими руками, проявив фантазию и воображение, создать красивую фасадную композицию, будь то клумба, палисадник или цветочное оформление территории. Также Вы можете применить свое фото-мастерство и оригинальность и участвовать в данном конкурсе в категории «Лучший цветочный кадр».
        </p>
      </div>
      <div
        className="col-md-10 rules"
      >
        <h2
          className="section_title"
        >
          ПРАВИЛА КОНКУРСА
        </h2>
        <p
          className="section_text"
        >
          Из перечня всех предлагаемых номинаций, находящихся на нашем сайте, Вы сможете выбрать любую, которая Вам по душе и непосредственно подходящую для Ваших условий, места проживания и территориального расположения. Далее Вам будет необходимо выполнить соответствующие действия для создания композиции или цветочного оформления, сделать несколько фотографий своего готового объекта и прикрепить к
           
          <a
            href="/#section-send-request"
          >
            заявке участника
          </a>
          , которую также можно найти на сайте нашего конкурса. После одобрения конкурсной комиссией заявки, за неё можно будет проголосовать, но вносить в неё изменения будет уже невозможно.
        </p>
        <p
          className="section_text"
        >
          Оценка участников будет осуществляется посредством народного голосования, главным жюри которого станут жители и гости города Казань. Вы сможете отслеживать количество набранных Вами голосов и всегда быть в курсе происходящего.
        </p>
        <p
          className="section_text"
        >
          Стоит отметить, что участниками Конкурса, помимо жителей Казани, могут стать разного рода предприятия, организации, а также учреждения различных организационно-правовых форм. Словом – все те, кто хочет внести свой бесценный вклад в улучшение внешнего облика нашего города.
        </p>
        <p
          className="section_text"
        >
          Подводить итоги голосования будет городская конкурсная комиссия, которая и определит впоследствии победителей конкурса. Авторы лучших работ станут обладателями ценных призов и подарков. Желаем успехов!
        </p>
      </div>
    </RuleText>
    <div
      className="row rules_shedule"
    >
      <div
        className="col-md-10 shedule_svg"
      >
        <img
          alt="shedule"
          className="img-fluid"
          src="../../images/shedule_svg.svg"
        />
      </div>
    </div>
    <div
      className="row rules_nominations"
    >
      <div
        className="col-md-10 nominations"
      >
        <h3
          className="nomination_subtitle"
        >
          Конкурс проводится по следующим номинациям:
        </h3>
        <div
          className="row"
        >
          <div
            className="col-md-6 rules_nominations_wrapper"
            key="3"
          >
            <span
              className="nominations_list"
            >
              Лучшее оформление фасада
            </span>
          </div>
          <div
            className="col-md-6 rules_nominations_wrapper"
            key="4"
          >
            <span
              className="nominations_list"
            >
              Лучшее оформление дворовой территории
            </span>
          </div>
          <div
            className="col-md-6 rules_nominations_wrapper"
            key="5"
          >
            <span
              className="nominations_list"
            >
              Лучшее оформление прилегающей территории для организаций и предприятий
            </span>
          </div>
          <div
            className="col-md-6 rules_nominations_wrapper"
            key="6"
          >
            <span
              className="nominations_list"
            >
              Самая оригинальная цветочная композиция
            </span>
          </div>
          <div
            className="col-md-6 rules_nominations_wrapper"
            key="7"
          >
            <span
              className="nominations_list"
            >
              Лучшее оформление территории, прилегающей к образовательному учреждению
            </span>
          </div>
          <div
            className="col-md-6 rules_nominations_wrapper"
            key="10"
          >
            <span
              className="nominations_list"
            >
              Лучший цветочный кадр в городской среде
            </span>
          </div>
        </div>
      </div>
      <hr />
    </div>
    <div
      className="row rules_promo"
    >
      <div
        className="col-md-6 promo"
      >
        <h3>
          Оценка участников Конкурса по соответствующим номинациям осуществляется народным голосованием среди жителей и гостей города Казань
        </h3>
      </div>
    </div>
  </section>
</SectionRules>
`;
